### ISSUE TITLE
Build fails with undefined `MEMBARRIER_CMD_QUERY`

### ISSUE THREAD

**Issue by zodiac** [facebook/hhvm#7296]
### HHVM Version

master (2618ad50b027a06d06ddb20288afdd769a18e06d)

building on 64-bit ubuntu 16.04

```
[ 25%] Building CXX object third-party/folly/CMakeFiles/folly.dir/src/folly/portability/Sockets.cpp.o
/home/xuanji/hhvm/third-party/folly/src/folly/portability/SysMembarrier.cpp: In function ‘bool folly::detail::sysMembarrierAvailable()’:
/home/xuanji/hhvm/third-party/folly/src/folly/portability/SysMembarrier.cpp:45:37: error: ‘MEMBARRIER_CMD_QUERY’ was not declared in this scope
   auto r = syscall(__NR_membarrier, MEMBARRIER_CMD_QUERY, /* flags = */ 0);
                                     ^
/home/xuanji/hhvm/third-party/folly/src/folly/portability/SysMembarrier.cpp:50:14: error: ‘MEMBARRIER_CMD_SHARED’ was not declared in this scope
   return r & MEMBARRIER_CMD_SHARED;
              ^
/home/xuanji/hhvm/third-party/folly/src/folly/portability/SysMembarrier.cpp: In function ‘int folly::detail::sysMembarrier()’:
/home/xuanji/hhvm/third-party/folly/src/folly/portability/SysMembarrier.cpp:58:35: error: ‘MEMBARRIER_CMD_SHARED’ was not declared in this scope
   return syscall(__NR_membarrier, MEMBARRIER_CMD_SHARED, /* flags = */ 0);
```

looks like it might just not be defining/including the right things here https://github.com/facebook/folly/blob/0137d629ca981aac0186bf63bea82c2d5b9d7832/folly/portability/SysMembarrier.cpp#L25-L34

**Comment by paulbiss** [comment]
cc @Orvid

**Comment by Orvid** [comment]
Blargh..... I blame whoever thought it was a good idea to mess with memory barriers....

**Comment by zodiac** [comment]
@Orvid any fix/workaround for it?

**Comment by Orvid** [comment]
A fix should be landing in master within an hour or so. It was due to an update in folly that was intended to support Xenial, but that added an extra configure check that I failed to add to the CMake files for HHVM.

**Comment by Orvid** [comment]
This should now be fixed with 795ca9ec4e42fad78c8ca2f7c87c9bd6a8129080 and d606b1c7848ef9964843b5643b285a110a906e77.

**Comment by zodiac** [comment]
can confirm that this is fixed for me
